### ISSUE TITLE
Why dropout was forced in training mode? 

### ISSUE THREAD

**Issue by kevinmtian** [NVIDIA/tacotron2#324]
https://github.com/NVIDIA/tacotron2/blob/master/model.py#L99

Should I disable dropout during inference? I tried so but it got very big loss (20ish) when I evaluate the checkpoint on text-audio pair from LJspeech. 

```
class Prenet(nn.Module):
    def __init__(self, in_dim, sizes):
        super(Prenet, self).__init__()
        in_sizes = [in_dim] + sizes[:-1]
        self.layers = nn.ModuleList(
            [LinearNorm(in_size, out_size, bias=False)
             for (in_size, out_size) in zip(in_sizes, sizes)])

    def forward(self, x):
        for linear in self.layers:
            x = F.dropout(F.relu(linear(x)), p=0.5, training=True)
        return x

```

**Comment by CookiePPP** [comment]
https://arxiv.org/pdf/1712.05884.pdf

>  In order to introduce output variation at inference time, dropout with probability 0.5 is applied only to layers in the pre-net of the autoregressive decoder

**Comment by kevinmtian** [comment]
Thanks @CookiePPP ! May I learn why disabling dropout during inference would produce such a different result? i.e. setting `x = F.dropout(F.relu(linear(x)), p=0.5, training=False)` when generating lmel frames from text and shifted lmel frames (teacher forcing). 
With dropout enabled; I was able to get MSE loss (the tacotron2 criterion) of 0.6 in one of the text-audio pair in test set; yet disabling dropout would produce a loss of nearly 20. 
Would we expect dropout layers behave differently in tacotron2 vs. other models? 
Thank you!

**Comment by kevinmtian** [comment]
Found several suggestions in another discussion #247 closing now and will reopen if I find something
